src/python/pants/base/workunit.py

```diff
@@ -146,9 +146,13 @@ def start(self, start_time=None):
   def end(self):
     """Mark the time at which this workunit ended."""
     self.end_time = time.time()
+
+    return self.path(), self.duration(), self._self_time(), self.has_label(WorkUnitLabel.TOOL)
+
+  def close_outputs(self):
```

Comment:
`def cleanup()` might be a more evocative name that more clearly demonstrates where this belongs in the workunit callback lifecycle.